<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!-- Common parameter file (provides max_vel, limit.* params) -->
  <arg name="common_param_path" default="$(find-pkg-share autoware_core_planning)/config/common.param.yaml"/>

  <!-- Main parameter file -->
  <arg name="minimum_rule_base_planner_param_path" default="$(find-pkg-share autoware_minimum_rule_based_planner)/config/minimum_rule_based_planner.param.yaml"/>

  <!-- Trajectory optimizer plugin parameter files -->
  <arg
    name="minimum_rule_base_planner_elastic_band_smoother_param_path"
    default="$(find-pkg-share autoware_minimum_rule_based_planner)/config/trajectory_optimizer_plugins/elastic_band_smoother.param.yaml"
  />

  <!-- Velocity smoother parameter file -->
  <arg name="minimum_rule_base_planner_jerk_filtered_smoother_param_path" default="$(find-pkg-share autoware_minimum_rule_based_planner)/config/velocity_smoother/jerk_filtered_smoother.param.yaml"/>

  <!-- Topic remappings -->
  <arg name="input_route" default="~/input/route"/>
  <arg name="input_vector_map" default="~/input/vector_map"/>
  <arg name="input_odometry" default="~/input/odometry"/>
  <arg name="input_acceleration" default="~/input/acceleration"/>
  <arg name="input_objects" default="~/input/objects"/>
  <arg name="output_candidate_trajectories" default="~/output/candidate_trajectories"/>

  <!-- Modifier plugins -->
  <arg name="launch_obstacle_stop" default="true"/>
  <arg name="launch_modules_list_end" default="&quot;&quot;]"/>
  <arg name="minimum_rule_base_planner_trajectory_modifier_launch_modules" default="["/>
  <let
    name="minimum_rule_base_planner_trajectory_modifier_launch_modules"
    value="$(eval &quot;'$(var minimum_rule_base_planner_trajectory_modifier_launch_modules)' + 'autoware::minimum_rule_based_planner::plugin::ObstacleStop, '&quot;)"
    if="$(var launch_obstacle_stop)"
  />
  <let
    name="minimum_rule_base_planner_trajectory_modifier_launch_modules"
    value="$(eval &quot;'$(var minimum_rule_base_planner_trajectory_modifier_launch_modules)' + '$(var launch_modules_list_end)'&quot;)"
  />

  <node pkg="autoware_minimum_rule_based_planner" exec="autoware_minimum_rule_based_planner_node" name="minimum_rule_based_planner_node" output="both">
    <!-- Load parameter files -->
    <param from="$(var common_param_path)"/>
    <param from="$(var minimum_rule_base_planner_param_path)" allow_substs="true"/>
    <param from="$(var minimum_rule_base_planner_elastic_band_smoother_param_path)" allow_substs="true"/>
    <param from="$(var minimum_rule_base_planner_jerk_filtered_smoother_param_path)" allow_substs="true"/>
    <param name="modifier_launch_modules" value="$(var minimum_rule_base_planner_trajectory_modifier_launch_modules)"/>

    <!-- Topic remappings -->
    <remap from="~/input/route" to="$(var input_route)"/>
    <remap from="~/input/vector_map" to="$(var input_vector_map)"/>
    <remap from="~/input/odometry" to="$(var input_odometry)"/>
    <remap from="~/input/acceleration" to="$(var input_acceleration)"/>
    <remap from="~/input/objects" to="$(var input_objects)"/>
    <remap from="~/output/candidate_trajectories" to="$(var output_candidate_trajectories)"/>
  </node>
</launch>
